### PR TITLE
Usage message for usability 

### PR DESCRIPTION
**Pull request by 0culus**
Also added gitignore.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (cdacamar) [inline .gitignore]: This file needs to go completely.  I don't want a .gitignore until the build system is _really_ solidified.

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
   set(RELEASE_FLAGS, "/GS /GL /W3 /Gy /Zi /O2")
   set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 else()
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
```
Comment (cdacamar) [inline CMakeLists.txt]: This needs to be combined into `${DEBUG_FLAGS}` and `${RELEASE_FLAGS}`. This way we leave the intrinsic compiler flags alone and stick to configuration specific flags :smile:

File: main.cpp [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
 void dump_tokens(bf::lex::lexer& l);
 void dump_commands(const bf::prog::program& p);
+inline void dump_help(const char **argv) {
```
Comment (cdacamar) [inline main.cpp]: Can this be just a forward decl?  Let's move the decl above `dump_tokens` and it's implementation likewise above `dump_tokens` impl.

Comment (0culus) [inline main.cpp]: I don't see why not.

**Comment by cdacamar** [comment]
Let's squash all these commits but `239f840` and `acb4008`

File: CMakeLists.txt (hunk 2 above)
Comment (0culus) [inline CMakeLists.txt]: Well, it doesn't work. If I add it to those variables, it doesn't seem to pick up the flag, cue a vomit of warnings about c++11 features.

File: .gitignore (hunk 1 above)
Comment (0culus) [inline .gitignore]: I'll gitignore it in my local dir.

File: CMakeLists.txt [hunk 4]
```diff
@@ -5,22 +5,23 @@ project(brainfuck_cpp)
...
+  set(RELEASE_FLAGS "/GS /GL /W3 /Gy /Zi /O2 /MD")
   set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 else()
-  set(DEBUG_FLAGS, "-g -O0 -Wall")
-  set(RELEASE_FLAGS, "-O3")
+  set(DEBUG_FLAGS "-g -O0 -Wall -std=c++14")
+  set(RELEASE_FLAGS "-O3 -std=c++14")
 endif()
 
+message(STATUS "flags = " ${DEBUG_FLAGS})
```
Comment (cdacamar) [inline CMakeLists.txt]: Let's remove this

File: CMakeLists.txt [hunk 5]
```diff
@@ -5,22 +5,23 @@ project(brainfuck_cpp)
...
-  set(DEBUG_FLAGS, "/GS /W3 /Gm /Zl /Od /Oy-")
-  set(RELEASE_FLAGS, "/GS /GL /W3 /Gy /Zi /O2")
+  set(DEBUG_FLAGS "/GS /W3 /Gm- /Zi /Od /Oy- /Ob0 /D\"_DEBUG\" /MDd")
+  set(RELEASE_FLAGS "/GS /GL /W3 /Gy /Zi /O2 /MD")
   set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 else()
-  set(DEBUG_FLAGS, "-g -O0 -Wall")
-  set(RELEASE_FLAGS, "-O3")
+  set(DEBUG_FLAGS "-g -O0 -Wall -std=c++14")
+  set(RELEASE_FLAGS "-O3 -std=c++14")
```
Comment (cdacamar) [inline CMakeLists.txt]: add `-Wall` here

File: CMakeLists.txt (hunk 4 above)
Comment (0culus) [inline CMakeLists.txt]: See c2014a4983a80f6ed2642429efff2f11502139ed

File: CMakeLists.txt (hunk 5 above)
Comment (0culus) [inline CMakeLists.txt]: See 1bbc42864ffc7be48419a3c8dba2473187d92599

**Comment by cdacamar** [comment]
:+1: LGTM

**Pull request merged**